File: bmodle/src/main/java/com/hong_world/bmodle/modle/local/TasksLocalDataSource.java
package com.hong_world.bmodle.modle.local;

import com.hong_world.bmodle.modle.TasksDataSource;

import io.reactivex.Observable;

/**
 * Date: 2018/8/13. 10:43
 * Author: hong_world
 * Description:
 * Version:
 */
public class TasksLocalDataSource implements TasksDataSource {
    private static TasksLocalDataSource INSTANCE;

    private TasksLocalDataSource() {
    }

    public static TasksLocalDataSource getInstance() {
        if (INSTANCE == null) {
            INSTANCE = new TasksLocalDataSource();
        }
        return INSTANCE;
    }

    @Override
    public Observable getFeedArticleList(int num) {
        return null;
    }
}

File: kotlin-module/src/main/java/com/hong_world/kotlin_module/modle/local/TasksLocalDataSource.kt
package com.hong_world.kotlin_module.modle.local

import com.hong_world.kotlin_module.bean.FeedArticleListData
import com.hong_world.kotlin_module.modle.BaseTasksDataSource
import io.reactivex.Observable

/**
 * Date: 2018/9/25. 16:22
 * Author: hong_world
 * Description:
 * Version:
 */
class TasksLocalDataSource : BaseTasksDataSource {


    private constructor()

    companion object {
        @JvmStatic
        fun getInstance() = Holder.INSTANCE
    }

    private object Holder {
        val INSTANCE = TasksLocalDataSource()
    }

    override fun getFeedArticleList(num: Int): Observable<FeedArticleListData>? = null
}